### PR TITLE
Make ksql-server customizable for Docker environment 

### PR DESCRIPTION
**Pull request by xmlking**
This PR allows us to run `ksql-server` with custom CLI parameters i.e.,  `--properties-file, --bootstrap-server, --command-topic-suffix, --service-id` or via  environment variables
`STREAMS_BOOTSTRAP_SERVERS`, `LISTENERS_CONFIG`

For example we can use  environment variables with docker-compose to overwrite build-in properties:

```yml
 ksql-server:
    image: confluent/ksql-cli
    ports:
      - 9098:9098
    networks:
      - reactive-network
    links:
      - kafka
    depends_on:
      - kafka
    environment:
      KSQL_CONFIG_DIR: "/etc/ksql"
      KSQL_LOG4J_OPTS: "-Dlog4j.configuration=file:/etc/ksql/log4j.properties"
      STREAMS_BOOTSTRAP_SERVERS: kafka:9092
    entrypoint: 'ksql-server-start'
```

### CODE REVIEW

File: ksql-core/src/main/java/io/confluent/ksql/util/KsqlConfig.java [hunk 1]
```diff
@@ -28,7 +28,7 @@
 
 public class KsqlConfig extends AbstractConfig {
 
-  public static final String KSQL_TIMESTAMP_COLUMN_INDEX = "ksq.timestamp.column.index";
+  public static final String KSQL_TIMESTAMP_COLUMN_INDEX = "ksql.timestamp.column.index";
```
Comment (miguno) [inline ksql-core/src/main/java/io/confluent/ksql/util/KsqlConfig.java]: This bug fix is unrelated to this PR but might still be worth getting into the code.

/cc @hjafarpour @bluemonk3y .

**Comment by xmlking** [comment]
Not needed based on the direction development is going

**Comment by apurvam** [comment]
Thanks @xmlking . Closing this out.

**Comment by miguno** [comment]
FYI: We have provided functionality to customize KSQL for Docker environments via https://github.com/confluentinc/ksql/pull/820.

**Pull request closed without merging**